aips/aip-x.md

```diff
@@ -0,0 +1,158 @@
+---
+aip: x
+title: Aptos Token Objects Framework Update
+author: johnchanguk
+discussions-to (*optional):
+Status: In Review
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Framework
+created: 03/23/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Aptos Token Objects Framework Update
+
+## Summary
+
+This AIP aims to extend the `collection`, `token` and `property_map` modules from the `aptos-token-objects` framework. Reasons for this are to introduce extensibility and customization to collection and token objects.
+
+## Motivation
+
+Currently there is no way of changing the name or max supply of a collection, nor create a token with the token and custom seed provided. Reasons for wanting this is for example:
+
+1. Some users may not want to have on chain data publicly visible, such as the collection name during collection creation. Being able to set this at a later date allows users to confidently deploy collections without leaking any sensitive information.
```

Comment:
I was reminded why this isn't so great and needs more thought:
* https://github.com/aptos-labs/aptos-core/blob/5a82310b900d65257edb7969a358580bc2af453b/aptos-move/framework/aptos-token-objects/sources/token.move#L247
* https://github.com/aptos-labs/aptos-core/blob/5a82310b900d65257edb7969a358580bc2af453b/aptos-move/framework/aptos-token-objects/sources/token.move#L268

by enabling the users to move the name, we kinda break collection accesses and have to remember a legacy name. It is a bit weird... thoughts on correctness and usability if we do that? do we want to add a few more functions that allow the devs to be explicit on collection address?

Reply:
That's the assumption to begin with as when we first created you told us that the name could be changed. We built indexing with this in mind and I'm pretty sure that sdk reads from indexer.

Also 99.9% collections won't do this. The worst case is that the collections that do do this will have to figure out how to manage their own collection id.

Reply:
This is a risk that needs to be explicitly called out in this doc, in the code, and our documentation. Also it is great if the indexer tracks the original collection name and we add APIs that enable us to create new tokens given a collection id.

Reply:
If we give something really risky, we should have clarity on how users can manage this in a reasonable fashion.

Reply:
Token creation functions which take in `Object<Collection>` will be created, as this solves this issue as discussed!